zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/UserTaskState.java

```diff
@@ -21,6 +23,8 @@ public interface UserTaskState {
 
   UserTaskIntermediateStateValue getIntermediateState(final long userTaskKey);
 
+  Optional<UserTaskRecordRequestMetadata> getRecordRequestMetadata(final long userTaskKey);
```

Comment:
🔧 As this returns an optional, we could consider naming this `findRecordRequestMetadata` to clarify that it could optionally return a value or not.

Reply:
Yeah, I've changed the method to return `Optional` just before committing the first commit 🙈.
Thanks Nico for highlighting this! I’ve renamed it to `findRecordRequestMetadata` 👍.